pipeline_dp/budget_accounting.py

```diff
@@ -113,10 +114,10 @@ def compute_budgets(self):
 class MechanismSpec:
     """Specifies the parameters for a mechanism.
 
-    NoiseKind defines the kind of noise distribution.
+    MechanismType defines the kind of noise distribution.
```

Comment:
defines the type of the mechanism ...

Reply:
👍 I also changed all instances of `noise_kind` etc. to mechanism type